### ISSUE TITLE
Spike Count Estimate S

### ISSUE THREAD

**Issue by danachang** [zhoupc/CNMF_E#39]
Hi, From Wiki and the paper, it suggests `S(i)` is the inferred number of spike events at i-th frame. I am thinking to use `S(i)` as a proxy to calculate firing rate. However, in CalmAn (Python package implementation of CNMF), it is emphasized that S cannot be directly used as number of spikes or spiking rate, and instead, it is just loosely proportional to spiking rate. It suggests people to use dF_F (`F_dff`) to deconvolve and further set some amplitude threshold. I also have seen papers taking CNMFe results and treat each `S(i)` as a potential spiking event and ignore the amplitude while calculating firing rate (ref: https://www.biorxiv.org/content/10.1101/292953v3.full.pdf). I'm thus curious to know which is the right way to do? Treat each `S(i)` as one spike event, or use the number of `S(i)` as spiking rate? Thanks a lot!

**Comment by zhoupc** [comment]
Thanks, @danachang 

first of all, it's correct that ‘S cannot be directly used as the number of spikes or spiking rate’. There are two reasons: (1) the [ca2+] increase is not linear with spike counts; (2) the fluorescence signal is proportional to [ca2+] increase, but this coefficient depends on the indicator, optical gain, neurons' spatial footprints, etc. 

Reason (2) suggests a trivial coefficient between the fluorescence signal DF and [ca2+] increase. To remove this statistical identifiability, we tend to normalize DF with either F0 or its noise level. That's the point of using DF/F suggested by CaImAn. 

In the deconvolution model of dF/F, S is explained as spike count in early literature (e.g., Vogelstein 2009 & 2010), where single spikes were recorded in those experiments. while in these days, the calcium transients may contain a burst of spikes. I think the appropriate explanation of those S should be [ca2+] increase. This increase may be a nonlinear function of spike count (firing rate), that's why CaImAn said ' it is just loosely proportional to spiking rate.'. 

Second, The S(i) in CNMF-E result is the same as the deconvolution result of 'dF_F (F_dff)' as suggested by CaImAn. The only difference is just trivial scaling. 

so in summary, there is no contradictory statements between cnmf-e wiki and caiman. I misled you because of the inaccurate statements. regarding your question, S[i] is loosely proportional to spiking rate for the i-th neuron.

**Comment by danachang** [comment]
@zhoupc Thank you very much for these detailed clarification!
